### PR TITLE
tests/gnrc_ipv6_ext_frag: fetch Interface IDs dynamically

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
Currently the test script for `tests/gnrc_ipv6_ext_frag` assumes a certain order and interface ID for the underlying application which provides one for a mocked transfer and one ethos interface to interact with the host-side scripting. However, ever since eb73515e2cfdf5e42c84dfd53489c0e63dc6f00b got merged (and probably others) got merged, that order and numbering is not true anymore, making the test fail at the moment.

This fixes the test to fetch the interface identifiers based on the Ethernet address of the interface (for ethos and assume the other to be the mock interface) and uses those identifiers throughout the script instead of the constant once.

### Testing procedure
```sh
BOARD=samr21-xpro make -C tests/gnrc_ipv6_ext_frag/ -j flash test-as-root
```

currently fails on master and in release branches `2020.10-branch` and `2021.01-branch`. With this fix, it will pass.

### Issues/PRs references
None

### CODE REVIEW

File: tests/gnrc_ipv6_ext_frag/tests-as-root/01-run.py [hunk 1]
```diff
@@ -173,10 +173,38 @@ def test_ipv6_ext_frag_shell_test_1(child, s, iface, ll_dst):
...
+        if match:
+            # interface has a hardware address
+            ethos_id = int(child.match.group(1))
+            hwaddr = match.group(1)
+            # consume MTU for later calls of `ifconfig {mock_id}`
+            child.expect(r"MTU:(\d+)")
+        else:
+            mock_id = int(child.match.group(1))
+            # consume MTU for later calls of `ifconfig {mock_id}`
+            child.expect(r"MTU:(\d+)")
```
Comment (aabadie) [inline tests/gnrc_ipv6_ext_frag/tests-as-root/01-run.py:195]: small improvement proposal:

```suggestion
        if match is not None:
            # interface has a hardware address
            ethos_id = int(child.match.group(1))
            hwaddr = match.group(1)
        else:
            mock_id = int(child.match.group(1))
        # consume MTU for later calls of `ifconfig {mock_id}`
        child.expect(r"MTU:(\d+)")
```

Comment (miri64) [reply tests/gnrc_ipv6_ext_frag/tests-as-root/01-run.py]: Taken :-)

**Comment by jia200x** [comment]
it doesn't work for me, it doesn't find `testrunner` for some reason :/

```
[sudo] password for jialamos: 
make[1]: Nothing to be done for 'all'.
Traceback (most recent call last):
  File "/home/jialamos/Development/RIOT/tests/gnrc_ipv6_ext_frag/tests-as-root/01-run.py", line 19, in <module>
    from testrunner import run, check_unittests
ModuleNotFoundError: No module named 'testrunner'
make: *** [/home/jialamos/Development/RIOT/makefiles/tests/tests.inc.mk:35: test-as-root] Error 1
```

@aabadie does it work for you?

**Comment by miri64** [comment]
> it doesn't work for me, it doesn't find `testrunner` for some reason :/
> 
> ```
> [sudo] password for jialamos: 
> make[1]: Nothing to be done for 'all'.
> Traceback (most recent call last):
>   File "/home/jialamos/Development/RIOT/tests/gnrc_ipv6_ext_frag/tests-as-root/01-run.py", line 19, in <module>
>     from testrunner import run, check_unittests
> ModuleNotFoundError: No module named 'testrunner'
> make: *** [/home/jialamos/Development/RIOT/makefiles/tests/tests.inc.mk:35: test-as-root] Error 1
> ```
> 
> @aabadie does it work for you?

What's your environment, @jia200x? Operating system, environment variables, etc...

**Comment by jia200x** [comment]
Well, I realized PYTHONPATH doesn't get too the script even with `sudo -E` :/.

**Comment by jia200x** [comment]
well, I had to hardcode the patch with `sys.path.append` and this PR works at least:

```
~/Development/RIOT/tests/gnrc_ipv6_ext_frag/ BOARD=samr21-xpro make test-as-root
make[1]: Nothing to be done for 'all'.
................SUCCESS
Skipping datagram reception tests due to ethos bug.
```

**Comment by jia200x** [comment]
> well, I had to hardcode the patch with sys.path.append and this PR works at least:

Given that, I'm fine with approving this PR. This won't trigger that issue in the release branch

**Comment by jia200x** [comment]
> Given that, I'm fine with approving this PR. This won't trigger that issue in the release branch

Hmmm, but we cannot backport it :/... there's no test-as-root there

**Comment by miri64** [comment]
> > Given that, I'm fine with approving this PR. This won't trigger that issue in the release branch
> 
> Hmmm, but we cannot backport it :/... there's no test-as-root there

Yes we can. The script was just moved from `tests`. That's where I did the rewrites actually and git was smart enough to rebase it to master without any conflicts. I expect the same for the reverse.

**Review by jia200x — APPROVED** [review]
Ok, then ACK. I confirm this fixes the issue (although I couldn't run it with test-as-root).

**Comment by jia200x** [comment]
please squash!

**Comment by miri64** [comment]
> Well, I realized PYTHONPATH doesn't get too the script even with `sudo -E` :/.

But the `PYTHONPATH` is provided with the target:

https://github.com/RIOT-OS/RIOT/blob/b3ffb690b1e3b100684e9531ec91c68d16818cf6/makefiles/tests/tests.inc.mk#L36

**Comment by jia200x** [comment]
as discussed offline, this is what happens with my setup:
```
$RIOTBASE/tests/gnrc_ipv6_ext_frag/ sudo -E PYTHONPATH=testValue bash -c 'echo $PYTHONPATH'

/usr/local/lib/python3.9/dist-packages:/usr/local/lib/python3.9/site-packages
```

**Comment by aabadie** [comment]
Are exporting PYTHONPATH from a bashrc or something ?

**Comment by miri64** [comment]
Squashed!

**Comment by jia200x** [comment]
> Are you exporting PYTHONPATH from a bashrc or something ?

Nope...

```
~/Development/RIOT/tests/gnrc_ipv6_ext_frag/ env | grep PYTHON
/* crickets */
```

**Comment by miri64** [comment]
Backport provided in #15858

**Pull request merged**